api/src/main/java/org/commonjava/indy/data/DefaultStoreValidator.java

```diff
@@ -0,0 +1,171 @@
+                            );
+                        }
+                    }
+                }
+                // Execute HTTP GET & HEAD requests in separate thread pool from executor service
+                Future<Integer> httpGetStatus = executeGetHttp(new HttpGet(remoteUrl.get().toURI()), httpRequestsLatch);
+                Future<Integer> httpHeadStatus = executeHeadHttp(new HttpHead(remoteUrl.get().toURI()), httpRequestsLatch);
+                // Waiting for Http GET & HEAD Request Executor tasks to finish
+                httpRequestsLatch.await();
+                errors.put(StoreValidationConstants.HTTP_GET_STATUS, httpGetStatus.get().toString());
+                errors.put(StoreValidationConstants.HTTP_HEAD_STATUS, httpHeadStatus.get().toString());
+                if(!remoteUrl.get().getProtocol().equalsIgnoreCase(StoreValidationConstants.HTTPS)) {
+                    errors.put(StoreValidationConstants.HTTP_PROTOCOL, remoteUrl.get().getProtocol());
+                }
+                // Check for Sucessfull Validation
+                if (httpGetStatus.get() < 400 && httpHeadStatus.get() < 400) {
+                    LOGGER.warn("=> Success HTTP GET and HEAD Response from Remote Repository: " + remoteUrl.get());
+                    return new ArtifactStoreValidateData
+                        .Builder(remoteRepository.getKey())
+                        .setRepositoryUrl(remoteUrl.get().toExternalForm())
+                        .setValid(true)
+                        .setErrors(errors)
+                        .build();
+                }
+            }
+        }
+        catch (MalformedURLException mue) {
+            LOGGER.error("=> Mailformed URL: ", mue);
+            errors.put(StoreValidationConstants.MAILFORMED_URL,mue.getMessage());
+            return new ArtifactStoreValidateData
+                .Builder(artifactStore.getKey())
+                .setRepositoryUrl( remoteUrl.get().toExternalForm() )
+                .setErrors(errors)
+                .build();
+        }
+        catch (Exception e) {
+            LOGGER.error(" => Not Valid Remote Repository, \n => Exception: " + e);
+            errors.put(StoreValidationConstants.GENERAL, e.getMessage());
+            return new ArtifactStoreValidateData
+                .Builder(artifactStore.getKey())
+                .setRepositoryUrl( remoteUrl.get().toExternalForm() )
+                .setErrors(errors)
+                .build();
+        }
+        if (artifactStore instanceof RemoteRepository) {
+            return new ArtifactStoreValidateData
+                .Builder(artifactStore.getKey())
+                .setRepositoryUrl(((RemoteRepository) artifactStore).getUrl())
+                .setErrors(errors)
+                .build();
+        }
+        else {
+            return new ArtifactStoreValidateData
+                .Builder(artifactStore.getKey())
```

Comment:
We should call `.setValid(true)` for these non-remote stores, since we didn't have any rules to invalidate them.